### PR TITLE
tbb: 2021.8.0 -> 2021.11.0

### LINKED ISSUE #301387: mold with libtbb version 2021.9.0 or older is unstable
### Describe the bug

The current `mold` package sometimes causes build failures due to instability (or warnings related to known instability).

### Steps To Reproduce

For example, trying to build `hyprland` with `mold` in the following fashion:

```nix
nixpkgs.overlays = [
  (_final: prev: {
    hyprland = prev.hyprland.override {
      stdenv = prev.useMoldLinker prev.llvmPackages.stdenv;
    };
  })
];
```

leads to the following output in `meson-log.txt`

```
mold: aborted
mold: mold with libtbb version 2021.9.0 or older is known to be unstable under heavy load. Your libtbb version is 2021.8. Please upgrade your libtbb library and try again.
mold: aborted
mold: mold with libtbb version 2021.9.0 or older is known to be unstable under heavy load. Your libtbb version is 2021.8. Please upgrade your libtbb library and try again.
/nix/store/84fxmapz4q1i6xjfsm7ml7g62i49r7wf-ccache-links-wrapper-4.9.1/bin/ld.mold: line 269:    53 Aborted                 (core dumped) /nix/store/b7hb6xpazdnidp0s87fain2im6lk4f4r-mold-2.30.0/bin/ld.mold ${extraBefore+"${extraBefore[@]}"} ${params+"${params[@]}"} ${extraAfter+"${extraAfter[@]}"}
clang: error: linker command failed with exit code 134 (use -v to see invocation)
```

### Expected behavior

`mold` should successfully link without issue.

### Notify maintainers

@azahi @paveloom

### LINKED ISSUE #301390: Update request: tbb: 2021.8.0 → 2021.11.0
- Package name: tbb
- Latest released version: [`2021.11.0`](https://github.com/oneapi-src/oneTBB/releases/tag/v2021.11.0)
- Current version on the unstable channel: `2021.8.0`
- Current version on the stable/release channel: `2021.8.0`
- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Additional Notes**

Updating `tbb` to `2021.11.0` fixes `mold`-related issue https://github.com/NixOS/nixpkgs/issues/301387

**Notify maintainers**

@thoughtpolice @hesiod

### PR DESCRIPTION
**Pull request by silvanshade**
Fixes #301390

## Description of changes

Updates `tbb` to `2021.11.0`.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

@thoughtpolice @hesiod

### CODE REVIEW

**Review by wegank — DISMISSED** [review]
Be careful not to bump `tbb_2021_8` from 2021.8.0 to 2021.11.0: https://github.com/NixOS/nixpkgs/blob/ae8f15ae441a54f9579c94186ab6d715fb9266c9/pkgs/top-level/all-packages.nix#L17093

Instead, try something like `tbb_2021_11: init at 2021.11.0`.

**Comment by silvanshade** [comment]
> Be careful not to bump `tbb_2021_8` from 2021.8.0 to 2021.11.0: https://github.com/NixOS/nixpkgs/blob/ae8f15ae441a54f9579c94186ab6d715fb9266c9/pkgs/top-level/all-packages.nix#L17093
> 
> Instead, try something like `tbb_2021_11: init at 2021.11.0`.

Thanks for the feedback. Is this what you had in mind?

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -17093,7 +17093,8 @@ with pkgs;
   swiPrologWithGui = swiProlog.override { withGui = true; };
 
   tbb_2020_3 = callPackage ../development/libraries/tbb/2020_3.nix { };
-  tbb_2021_8 = callPackage ../development/libraries/tbb { };
+  tbb_2021_8 = callPackage ../development/libraries/tbb/2021_8.nix { };
+  tbb_2021_11 = callPackage ../development/libraries/tbb { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:17097]: I've just successfull build all packages that depend on tbb_2021_8 and we can just upgrade them all. cctag is already broken on master but it fails in tests, not while building/linking.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by silvanshade** [comment]
@wegank @SuperSandro2000 I made the requested changes and renamed `tbb` tree-wide.

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 301391` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>2 packages failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>cctag: <a href="https://termbin.com/484a">plain log</a></li>
<li>super-slicer: <a href="https://termbin.com/8h33">plain log</a></li>
  </ul>
</details>
<details>

  <summary>3 packages failed to build and are new build failures:</summary>
  <ul>
<li>cctag.dev: <a href="https://termbin.com/opxi">plain log</a></li>
<li>cctag.lib: <a href="https://termbin.com/9vx7">plain log</a></li>
<li>super-slicer.debug: <a href="https://termbin.com/o0bg">plain log</a></li>
  </ul>
</details>
<details>
<summary>
21 packages built:
</summary>
<ul>
<li>bambu-studio</li>
<li>bambu-studio.debug</li>
<li>ikos</li>
<li>mold</li>
<li>mold-wrapped</li>
<li>openscad-unstable</li>
<li>orca-slicer</li>
<li>orca-slicer.debug</li>
<li>pdf4qt</li>
<li>prusa-slicer</li>
<li>prusa-slicer.debug</li>
<li>salmon</li>
<li>super-slicer-beta</li>
<li>super-slicer-beta.debug</li>
<li>super-slicer-latest</li>
<li>super-slicer-latest.debug</li>
<li>tbb_2021_11</li>
<li>tbb_2021_11.dev</li>
<li>ueberzugpp</li>
<li>universal-android-debloater</li>
<li>ytfzf</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
bambu-studio.debug:
<details>

got build log for '/nix/store/ag9vnc0ghcylsmzql2xxqsy3c6zw42j5-bambu-studio-01.09.00.60-debug' from 'daemon'
When evaluating attribute ‘bambu-studio.debug’:
warning: unused-argument
Unused argument: `qhull`.
Near pkgs/applications/misc/bambu-studio/default.nix:39:3:

```
   |
39 | , qhull
   |   ^
```
warning: unused-argument
Unused argument: `fetchpatch`.
Near pkgs/applications/misc/bambu-studio/default.nix:45:3:

```
   |
45 | , fetchpatch
   |   ^
```
</details>
orca-slicer.debug:
<details>

got build log for '/nix/store/ykcshww40nn9m4pfnrmjs1xqx2bhnrkh-orca-slicer-1.9.1-debug' from 'daemon'
When evaluating attribute ‘orca-slicer.debug’:
warning: unused-argument
Unused argument: `makeDesktopItem`.
Near pkgs/applications/misc/bambu-studio/orca-slicer.nix:1:25:

```
  |
1 | { lib, fetchFromGitHub, makeDesktopItem, bambu-studio }:
  |                         ^
```
</details>
orca-slicer:
<details>

got build log for '/nix/store/n6f3hvb7p5f8j1j0ljxbbn07x37zk16g-orca-slicer-1.9.1' from 'daemon'
When evaluating attribute ‘orca-slicer’:
warning: unused-argument
Unused argument: `makeDesktopItem`.
Near pkgs/applications/misc/bambu-studio/orca-slicer.nix:1:25:

```
  |
1 | { lib, fetchFromGitHub, makeDesktopItem, bambu-studio }:
  |                         ^
```
</details>
super-slicer-latest.debug:
<details>

got build log for '/nix/store/1ng6c8zxg4zyycmcambqm8g52q6wqchj-super-slicer-2.4.58.5-debug' from 'daemon'
When evaluating attribute ‘super-slicer-latest.debug’:
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern list\(APPEND\ wxWidgets_LIBRARIES\ libexpat\) doesn't match anything in file 'src/CMakeLists.txt'
Near pkgs/applications/misc/prusa-slicer/super-slicer.nix:83:

```
   |
83 |       inherit description;
   | ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern libexpat doesn't match anything in file 'src/libslic3r/CMakeLists.txt'
Near pkgs/applications/misc/prusa-slicer/super-slicer.nix:83:

```
   |
83 |       inherit description;
   | ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
</details>
super-slicer-latest:
<details>

got build log for '/nix/store/jrykac1aa39fc98yxcxpc3y88fvmkbl3-super-slicer-2.4.58.5' from 'daemon'
When evaluating attribute ‘super-slicer-latest’:
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern list\(APPEND\ wxWidgets_LIBRARIES\ libexpat\) doesn't match anything in file 'src/CMakeLists.txt'
Near pkgs/applications/misc/prusa-slicer/super-slicer.nix:83:

```
   |
83 |       inherit description;
   | ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern libexpat doesn't match anything in file 'src/libslic3r/CMakeLists.txt'
Near pkgs/applications/misc/prusa-slicer/super-slicer.nix:83:

```
   |
83 |       inherit description;
   | ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
</details>
</details>

**Pull request merged**